src/vmm/src/acpi/mod.rs

```diff
@@ -272,15 +273,18 @@ pub mod tests {
             resource_allocator: &mut vmm.resource_allocator,
         };
 
-        let mut sdt = MockSdt(vec![0; usize::try_from(SYSTEM_MEM_SIZE).unwrap()]);
+        let mut sdt = MockSdt(vec![
+            0;
+            usize::try_from(SYSTEM_MEM_SIZE - MMIO_LEN).unwrap()
+        ]);
         let err = writer.write_acpi_table(&mut sdt).unwrap_err();
         assert!(
             matches!(
                 err,
                 AcpiError::AcpiTables(acpi_tables::AcpiError::GuestMemory(
                     vm_memory::GuestMemoryError::PartialBuffer {
-                        expected: 263168,  // SYSTEM_MEM_SIZE
-                        completed: 259072  // SYSTEM_MEM_SIZE - 4096
+                        expected: 259072,  // SYSTEM_MEM_SIZE - MMIO_LEN
+                        completed: 254976, // SYSTEM_MEM_SIZE - MMIO_LEN * 2
```

Comment:
not related with your PR, but I should have probably used the expressions in the comments rather than the hardcoded numerics. If you feel like, could you please fix it in the next iteration?

Reply:
I don't think you can use the expression here, I think we have to use the actual values because we're inside a pattern match. We have to convert the constant values into `usize`, at which point the compiler no longer views them as constant, and therefore pattern matching won't actually match on the exact values, just the structure.

Reply:
oh ok, that explains why it was like that in the first place. Thanks for looking into it.